src/ghga_transpiler/config.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Module to process config file"""
+
+from collections import Counter
+from collections.abc import Callable
+
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    model_validator,
+)
+
+from .exceptions import DuplicatedName
+from .transformations import to_attributes, to_list, to_snake_case, to_snake_case_list
+
+
+class ColumnMeta(BaseModel):
+    """A data model for column properties"""
+
+    model_config = ConfigDict(populate_by_name=True, frozen=True)
+
+    sheet_name: str = Field(..., alias="sheet")
+    column_name: str = Field(..., alias="column")
+    multivalued: bool
+    type: str
+    ref_class: str | None
+    ref_id: str | None = Field(..., alias="ref_class_id_property")
+    enum: bool
+    required: bool
+
+    def transformation(self) -> Callable | None:
+        """Assigns transformation function based on column properties"""
+        if self.multivalued and self.enum:
+            return to_snake_case_list()
+        elif self.enum:
+            return to_snake_case()
+        elif self.multivalued and self.type == "object":
+            return to_attributes()
+        elif self.multivalued:
+            return to_list()
+        else:
+            return lambda value: value
+
+    def relation(self) -> bool:
+        """If a column is a relation column"""
+        if self.ref_class:  # noqa: SIM103
+            return True
+        return False
```

Comment:
```suggestion
    def is_relation(self) -> bool:
        """Return whether this is a relation column"""
        return bool(self.ref_class)
```

Makes it more obvious from the function name that it returns a bool.